### PR TITLE
Add sanity check for cml-publish with filesystem paths

### LINKED ISSUE #308: [cml-publish] If file does not exists says that mime type is wrong
Comming from  [Discord](https://discord.com/channels/485586884165107732/728693131557732403/763840404675756042)

### LINKED ISSUE #401: [Failed guessing mime type of file] Unable to publish ".png" image to the report.
Hi there ! 

# Description
Based on [the proposed by the DVCorg community MLOps tutorial](https://www.youtube.com/watch?v=9BgIDqAzfuA&list=PL7WG7YrwYcnDBDuCkFbcyjnZQrdskFsBz) I was attaching the calculated and saved `confusion_matrix.png` image to the report. But I continuously obtained the error: 
```bash
Error: Failed guessing mime type of file result/confusion_matrix.png
    at mime_type (/cml/src/utils.js:49:11)
Error: Process completed with exit code 1.

```

### The `.github/workflow/expirement.cml` file content:
```bash
name: experiment-workflow
on: [push]
jobs:
  run:
    runs-on: [ubuntu-latest]
    container: docker://dvcorg/cml-py3:latest
    steps:
      - uses: actions/checkout@v2
      - name: cml_run
        env:
          repo_token: ${{ secrets.GITHUB_TOKEN }}
        run: |

          pip install -r requirements.txt
          python expirement.py

          # CML REPORT
          echo "## Model Metrics" >> report.md
          cat results/metrics.txt >> report.md

          echo "## Confusion Matrix" >> report.md
          cml-publish results/confusion_matrix.png --md >> report.md

          cml-send-comment report.md
```

### The sample of the `expirement.py` file content:
```python
# imports
...

_FOLDER = Path('./results')
_METRICS = Path('metrics.txt')
_CONFUSION_MATRIX = Path('confusion_matrix.png')


shutil.rmtree(_FOLDER, ignore_errors=False)
Path(_FOLDER).mkdir(exist_ok=True)

# example of model output
target = [1, 1, 1, 0, 0, 1]
predicted = [1, 1, 1, 1, 1, 0]


results_dict = {
    'Precision': MetricsFactory.precision(target=target, predicted=predicted),
    'Recall': MetricsFactory.recall(target=target, predicted=predicted),
    'ROC AUC': MetricsFactory.roc_auc(target=target, predicted=predicted),
    'F1 Score': MetricsFactory.f1(target=target, predicted=predicted),
    'Confusion Matrix': MetricsFactory.confusion_matrix(target=target, predicted=predicted)
}

with open(_FOLDER / _METRICS, 'w') as f:
    f.write(str(results_dict))

_confusion_matrix = pd.DataFrame(results_dict['Confusion Matrix'])

plt.figure(figsize=(10, 7))
sn.heatmap(_confusion_matrix, annot=True)
plt.savefig(_FOLDER / _CONFUSION_MATRIX)
```

# Summary
The error denotes an issue with the `confusion_matrix.png` file. In the meantime I can open and validate the image exists and format is not corrupted.  
Could you please give a hint what's going wrong ? What did I miss if so ?

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Resolves #308 and probably also closes #401 by throwing a meaningful error for paths that don't exist, aren't files or can't be read.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -55,6 +55,14 @@ const mime_type = async (opts) => {
...
+    try {
+      await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
+    } catch (err) {
+      throw new Error(`Path ${path} does not exist or is not a readable file.`);
+    }
+  }
+
   const mime = await mime_type(opts);
   const data = path ? fs.createReadStream(path) : buffer;
   const size = path ? (await fs.promises.stat(path)).size : buffer.length;
```
Comment (0x2b3bfa0) [inline src/utils.js:68]: Indeed, running `stat` before `mime_type` wold grant us a similarly meaningful error with less code, but we wouldn't able to explicitly check if the path is a file nor if its contents are readable.

```suggestion
  const size = path ? (await fs.promises.stat(path)).size : buffer.length;
  const mime = await mime_type(opts);
  const data = path ? fs.createReadStream(path) : buffer;
```

**Comment by DavidGOrtega** [comment]
Let me check this. But the issue is that in the command line there is no difference between a buffer and a path. I mean you can do:

 - cml-publish path
 - cml-publish buffer

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega, both of my proposed solutions consider the buffer/path dichotomy, taking care only of the latter.

**Comment by DavidGOrtega** [comment]
Nice! Can we add a small test with a non existent file? Also with a a buffer would be awesome.

**Comment by 0x2b3bfa0** [comment]
I finally chose to call `stat` before calling `mime_type` so it acts as a meaningful safeguard. If you want to offer a custom but more broad error message, we can revert to the previous solution:

```javascript
if (path) {
  try {
    await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
  } catch (err) {
    throw new Error(`Path ${path} does not exist or is not a readable file.`);
  }
}
```

**Review by DavidGOrtega — COMMENTED** [review]
lgtm 👍

**Pull request merged**